brave/brave-core
# Add Rewards terms of service update notifications

Resolves https://github.com/brave/brave-browser/issues/35977

<img width="1149" alt="Screenshot 2024-03-01 at 3 55 52 PM" src="https://github.com/brave/brave-core/assets/5995084/d61c293a-c853-414b-b558-b01fb0e5824c">

<img width="950" alt="Screenshot 2024-03-01 at 3 26 10 PM" src="https://github.com/brave/brave-core/assets/5995084/d1d394d3-43c7-4afd-950c-8cafd3e207c7">

<img width="1149" alt="Screenshot 2024-03-01 at 3 26 23 PM" src="https://github.com/brave/brave-core/assets/5995084/1d660517-32d9-499a-8a4a-54b45a6e4b0b">

<img width="1149" alt="Screenshot 2024-03-01 at 3 26 30 PM" src="https://github.com/brave/brave-core/assets/5995084/7d1a3252-80db-43ee-9e41-b27edfd3cb3b">




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Feature Overview

* Given that the user has enabled Rewards,
* When the "tosVersion" number reported by the Rewards "parameters" endpoint is incremented,
* And the user's "tos_version" number stored in Preferences is less than that value,
* Then a ToS update notice should be displayed on the NTP, the Rewards panel, and the Rewards page.

### Test Steps

* With a new profile, enable Rewards.
* Close the browser.
* Increment the "tosVersion" field of the Rewards parameters endpoint.
  * The same effect can also be achieved by **decrementing** the `"tos_version"` value stored in `Preferences` (found under `"brave" >> "rewards"`.
* Restart the browser.
* Verify that the ToS update notice is displayed on the NTP, the Rewards page, and the Rewards panel.
* Verify that clicking the Terms of Service link opens a new tab with the correct URL.
* Verify that clicking the "reset" link shows the settings page with the "Reset Rewards" modal displayed.
* Click the accept button at any of the 3 locations.
* Verify that the ToS update notice is no longer displayed.

### Upgrade Case

If the "tosVersion" field returned from the "parameters" endpoint has a value of `1`, then the ToS update notice should not be displayed when the user upgrades their browser.

* With a profile created in a previous version, with Rewards enabled:
* Ensure that the "parameters" endpoint is returning `1` for the "tosVersion" field.
* Open the browser.
* Verify that the ToS update notice is not displayed.
